### PR TITLE
Remove therubyracer gem

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
Remove `therubyracer` gem from the LMS Gemfile.  Add `mini_racer` gem to the test group.

## WHY
Heroku [reports](https://devcenter.heroku.com/articles/rails-asset-pipeline#therubyracer) that therubyracer is no longer needed since "A version of Node is installed by the Ruby buildpack that will be used to compile your assets."  However, circleci cannot run tests due to "ExecJS::RuntimeUnavailable: Could not find a JavaScript runtime."   For this reason, we should use a more modern Gem (i.e. mini_racer) to satisfy the JS runtime requirement, while also potentially fixing an issue with bin/dev/boostrap.sh, `therubyracer` and Apple M1 chips.

NB: I'm pinning `mini_racer` to `0.3.1` since later versions either have an open [bug](https://github.com/rubyjs/mini_racer/issues/220) or require newer versions of Ruby.

## HOW
Update the Gemfile accordingly and run `bundle update`

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/LMS-Remove-therubyracer-ideal-or-fix-bootstrap-for-M1-Macs-f362f97698b448b0bfaf70b2e5608e4d

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  No, removal of redundant library.
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | N/A

### CODE REVIEW

**Review by dandrabik — APPROVED** [review]
Looks good to me! PRs like this are more about the testing, so I'll defer to you on that.

Thanks for cleaning this up!

**Comment by brendanshean** [comment]
I agree.  I'm in the process of fixing the ghost inspector staging tests.  Once those are passing I will merge this in.

**Pull request merged**
